src/plugins/intel_gpu/src/graph/impls/ocl/assign.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "assign_inst.h"
+#include "primitive_base.hpp"
+#include "impls/implementation_map.hpp"
+
+namespace cldnn {
+namespace ocl {
+
+struct assign_impl : public typed_primitive_impl_ocl<assign> {
+    using parent = typed_primitive_impl_ocl<assign>;
+    using parent::parent;
+
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<assign_impl>(*this);
+    }
+
+    event::ptr execute_impl(const std::vector<event::ptr>& events, assign_inst& instance) override {
+        const auto arg = instance.argument;
+        const auto variable_id = arg.variable_id;
+        auto& variable = instance.get_network().get_variable(variable_id);
+
+        if (variable.memory->get_layout() != arg.output_layout) {
+            CLDNN_ERROR_MESSAGE(instance.id(), "Layout mismatch");
+        }
+
+        std::vector<event::ptr> tmp_events{events};
+        auto& stream = instance.get_network().get_stream();
+
+        stream.enqueue_barrier(); // input must be ready before proceed
+
+        const auto ev_set_memory = variable.memory->copy_from(stream, instance.input_memory());
+        tmp_events.push_back(ev_set_memory);
+        variable.is_set = true;
```

Comment:
I have a question. Doesn't 'is_set' variable of Variable instance have to be able to be set back to false again for reset state? [state-api](https://docs.openvino.ai/latest/openvino_docs_OV_UG_network_state_intro.html#openvino-state-api)